pkg/controller/machinepool/azureactuator_test.go

```diff
@@ -153,8 +160,9 @@ func TestAzureActuator(t *testing.T) {
 			pool:              testAzurePool(),
 			mockAzureClient: func(mockCtrl *gomock.Controller, client *mockazure.MockClient) {
 				mockListResourceSKUs(mockCtrl, client, []string{"zone1", "zone2", "zone3"})
-				mockGetHyperVGeneration(mockCtrl, client, "V2")
+				mockGetVMCapabilities(mockCtrl, client, "V1,V2")
```

Comment:
Should there be a test case where only V1 exists? Similarly for only V2?

Reply:
Added tests for only V1 or only V2. As discussed offline, I think we need a FUP around this that future proofs the implementation with the expectation that V3 images will be introduced in the future.